beacon-chain/cache/depositcache/deposits_cache.go

```diff
@@ -169,10 +161,14 @@ func (dc *DepositCache) InsertFinalizedDeposits(ctx context.Context, eth1Deposit
 		}
 		insertIndex++
 	}
-
-	dc.finalizedDeposits = &FinalizedDeposits{
-		Deposits:        depositTrie,
-		MerkleTrieIndex: eth1DepositIndex,
+	tree, ok := depositTrie.(*trie.SparseMerkleTrie)
+	if !ok {
+		log.Error("Not a sparse merkle tree")
```

Comment:
Shouldn't this be an impossible condition ? Wouldn't it be better to actually return an error here

Reply:
You are right, that does seem weird.